src/wallet-user/trust-protection/VaultRecoveryProcessTab.tsx

```diff
@@ -0,0 +1,267 @@
+
+    const { api } = useLogionChain();
+    const { accounts, availableLegalOfficers, axiosFactory, refresh, colorTheme } = useCommonContext();
+    const { recoveredAddress, recoveryConfig } = useUserContext();
+    const [ recoveredCoinBalance, setRecoveredCoinBalance ] = useState<CoinBalance | null>(null);
+    const [ recoveredVaultAddress, setRecoveredVaultAddress ] = useState<string | null>(null)
+    const [ targetVaultAddress, setTargetVaultAddress ] = useState<string | null>(null)
+    const [ vaultBalances, setVaultBalances ] = useState<CoinBalance[] | null>(null)
+    const [ legalOfficer, setLegalOfficer ] = useState<string | null>(null)
+    const [ status, setStatus ] = useState<Status>(Status.IDLE)
+    const [ asyncSignAndSubmit, setAsyncSignAndSubmit ] = useState<AsyncSignAndSubmit>(null);
+    const [ signAndSubmitError, setSignAndSubmitError ] = useState<boolean>(false);
+    const [ candidates, setCandidates ] = useState<LegalOfficer[]>([]);
+
+    useEffect(() => {
+        if (candidates.length === 0 && recoveryConfig && availableLegalOfficers) {
+            setCandidates(availableLegalOfficers.filter(legalOfficer => recoveryConfig.legalOfficers.includes(legalOfficer.address)));
+        }
+    }, [ candidates, setCandidates, availableLegalOfficers, recoveryConfig ])
+
+    useEffect(() => {
+        if (recoveredVaultAddress === null && recoveredAddress && recoveryConfig) {
+            setRecoveredVaultAddress(getVaultAddress(recoveredAddress, recoveryConfig))
+        }
+    }, [ recoveredVaultAddress, setRecoveredVaultAddress, recoveredAddress, recoveryConfig ])
+
+    useEffect(() => {
+        if (targetVaultAddress === null && accounts && recoveryConfig) {
+            setTargetVaultAddress(getVaultAddress(accounts.current!.address, recoveryConfig))
+        }
+    }, [ targetVaultAddress, setTargetVaultAddress, accounts, recoveryConfig])
+
+    useEffect(() => {
+        if (vaultBalances === null && api !== null && recoveredVaultAddress) {
+            getBalances({ api, accountId: recoveredVaultAddress })
+                .then(balances => {
+                    setVaultBalances(balances.filter(balance => Number(balance.balance.toNumber()) > 0))
+                })
+        }
+    }, [ vaultBalances, setVaultBalances, recoveredVaultAddress, api ])
+
+    const { control, formState: { errors } } = useForm<FormValues>({
+        defaultValues: {
+            legalOfficer: null
+        }
+    });
+
+    const clearFormCallback = useCallback(() => {
+        setRecoveredCoinBalance(null);
+        setAsyncSignAndSubmit(null);
+        setSignAndSubmitError(false);
+        setLegalOfficer(null);
+        setStatus(Status.IDLE);
+    }, [ setRecoveredCoinBalance, setAsyncSignAndSubmit, setSignAndSubmitError ])
+
+    const amountToRecover = useMemo<PrefixedNumber>(() => recoveredCoinBalance?.balance ?
+            recoveredCoinBalance?.balance :
+            new PrefixedNumber("0", MILLI),
```

Comment:
Why MILLI and not NONE?

Reply:
That expression is mainly used during development phase, where you can hard-code a specific amount to transfer without emptying the account. I will fix.